src/openfermion/utils/_grid_test.py

```diff
@@ -57,3 +57,29 @@ def test_properties(self):
             (2, 1),
             (2, 2),
         ])
+
+    def test_repr(self):
+        self.assertEqual(
+            repr(Grid(dimensions=5, length=3, scale=0.5)),
+            'Grid(dimensions=5, length=3, scale=0.5)')
+
+    def test_str(self):
+        self.assertEqual(
+            str(Grid(dimensions=5, length=4, scale=0.5)),
+            '[-0.25, -0.125, 0.0, 0.125]**5')
+
+    def test_equality(self):
+        g = Grid(dimensions=5, length=3, scale=0.5)
+        g2 = Grid(dimensions=5, length=3, scale=0.5)
+
+        self.assertTrue(g == g2)
+        self.assertFalse(g != g2)
+        self.assertEqual(g, g2)
+        self.assertEqual(hash(g), hash(g2))
```

Comment:
Why does this work? I'm not used to hashes necessarily being equal for user defined classes.

Reply:
The two instances were defined with the same values, and the class supports value equality, therefore the hashes must be equal.

Reply:
Oh, I missed that you defined __hash__ for Grid. AFAIK Python's hash is basically blind to value equality, e.g. the following code will raise an error in general:

`class Integer(object):
	def __init__(self, i):
		self.i = i
	def __eq__(self, other):
		return self.i == other.i
a = Integer(3); b = Integer(3)
assert hash(a) == hash(b)`

Reply:
(Sorry for the lack of line breaks, I did my best.. :( )

Reply:
Use triple-mark to make code blocks. Triplet-mark then python to get highlighting.

![image](https://user-images.githubusercontent.com/79941/36928206-ad52eec6-1e38-11e8-9f2d-9841471246d9.png)

```python
def oh():
    pass
```